flink-runtime/src/main/java/org/apache/flink/runtime/taskmanager/Task.java

```diff
@@ -1081,23 +1081,23 @@ private boolean transitionState(
                         currentState,
                         newState);
                 LOG.debug(
-                        "{} ({}) switched from {} to {} due to CancelTaskException: {}",
+                        "{} ({}) switched from {} to {} due to CancelTaskException:",
                         taskNameWithSubtask,
                         executionId,
                         currentState,
                         newState,
-                        ExceptionUtils.stringifyException(cause));
```

Comment:
could we fix the duplicate exception as well as part of this change? The two log messages should be wrapped in a if/else block with `LOG.isDebugEnabled()` as the condition.